### DISCUSSION TITLE
[Q2 Vite] drop console config in build options

### DISCUSSION THREAD

**Discussion by ssuess** [quasarframework/quasar#13890] [CLI - PWA mode] [answered]
Following the quasar and vite docs I have this set:
```
build: {
  minify: 'terser',
    terserOptions: {
        compress: {
           drop_console: true,
           drop_debugger: true
        }
      }
```
but my builds are still showing console.log statements. Am I doing something wrong?

**Comment by ssuess** [comment]
have also tried:
```
minify: 'esbuild',
esbuild: {
  drop: ['console', 'debugger'],
},
```
which also does not work for me. :/

**Comment by metalsadman** [comment, marked as answer]
try it inside `extendViteConf`, as those property you are setting aren't found in the root `build` prop found here https://quasar.dev/quasar-cli-vite/quasar-config-js#build

so something like:
```js
// quasar.conf.js
...
  build: {
    ...
    extendViteConf(conf) {
      conf.esbuild = {
        drop: ['console', 'debugger']
      }
    }
...
```
https://vitejs.dev/config/#esbuild

**Reply by ssuess** [reply]
Any suggestions for getting this to work with terser? I have tried:
```
minify: 'terser',
extendViteConf (viteConf) {
        viteConf.terserOptions = {
          compress: {
            drop_console: true,
            drop_debugger: true
         }
        }
      },
```
as well as 
```
minify: 'terser',
extendViteConf (viteConf) {
        viteConf.terser = {
          compress: {
            drop_console: true,
            drop_debugger: true
         }
        }
      },
```
neither of which works. Thanks for any advice.

**Reply by metalsadman** [reply]
the former lgtm so no idea there, vite docs do say that terser is slower https://vitejs.dev/config/#build-minify.

**Reply by ssuess** [reply]
yeah, I read the docs which is how I got the format of the first one, but it does not work. I can't tell if this is a quasar specific failure or something else. And the only reason I want to use terser at all is because esbuild drops all kinds of css rules, including ones my app depends on for some of it's functionality.

**Reply by metalsadman** [reply]
try to set the `minify` inside the extendViteConf function as well.
```js
viteConf.build.minify = 'terser'
viteConf.build.terserOptions = {
  compress: {
    drop_console: true,
    drop_debugger: true
  }
}
```
if that still don't work, make a ticket.
edit: nvm above, just saw you already made one.

>And the only reason I want to use terser at all is because esbuild drops all kinds of css rules, including ones my app depends on for some of it's functionality.

is it the same issue in this ticket? https://github.com/evanw/esbuild/issues/1965

**Reply by ssuess** [reply]
I don't think mine is related to that esbuild ticket, it isn't related to `// comments` inside my `app.css` file

**Reply by ssuess** [reply]
However, one slightly different thing (perhaps) about my css file is that I declare `:root` vars, although not sure if that should matter.

**Reply by ssuess** [reply]
Well I'll be damned. Although I had no instance of `// comments` in my css file, I did have several of `/* this type of comment */` and going through and removing ALL of those seems to have fixed my output for esbuild! Thanks for pointing me to this!

**Reply by snowwolfjay** [reply]
can you try this,should work. just remember config the terserOption under the build scope at viteConf ha
quasar.config.js
```
 build:{
 // others
 minify: 'terser',
      // polyfillModulePreload: true,
      // distDir

extendViteConf(viteConf) {
  viteConf.build.terserOptions = viteConf.build.terserOptions || {};
  viteConf.build.terserOptions = {
    ...viteConf.build.terserOptions,
    compress: {
      ...viteConf.build.terserOptions.compress,
      drop_console: true,
    },
  };
 }
},
```
